Source/GTMSessionUploadFetcher.m

```diff
@@ -1239,75 +1296,64 @@ - (void)uploadNextChunkWithOffset:(int64_t)offset fetcherProperties:(NSDictionar
-                                          [self beginChunkFetcher:chunkFetcher offset:offset];
-                                          return;
-                                        }
-                                      }
-                                    }
-
-                                    if (chunkData == nil) {
-                                      NSError *responseError = chunkError;
-                                      if (!responseError) {
-                                        responseError =
-                                            [self uploadChunkUnavailableErrorWithDescription:
-                                                      @"chunkData is nil"];
-                                      }
-                                      [self invokeFinalCallbackWithData:nil
-                                                                  error:responseError
-                                               shouldInvalidateLocation:YES];
-                                      return;
-                                    }
-
-                                    BOOL didWriteFile = NO;
-                                    if (isUploadingFileURL) {
-                                      // Make a temporary file with the data subset.
-                                      NSString *tempName =
-                                          [NSString stringWithFormat:@"GTMUpload_temp_%@",
-                                                                     [[NSUUID UUID] UUIDString]];
-                                      NSString *tempPath = [NSTemporaryDirectory()
-                                          stringByAppendingPathComponent:tempName];
-                                      NSError *writeError;
-                                      didWriteFile = [chunkData writeToFile:tempPath
-                                                                    options:NSDataWritingAtomic
-                                                                      error:&writeError];
-                                      if (didWriteFile) {
-                                        chunkFetcher.bodyFileURL = [NSURL fileURLWithPath:tempPath];
-                                      } else {
-                                        GTMSESSION_LOG_DEBUG(@"writeToFile failed: %@\n%@",
-                                                             writeError, tempPath);
-                                      }
-                                    }
-                                    if (!didWriteFile) {
-                                      chunkFetcher.bodyData = [chunkData copy];
-                                    }
-                                    [self beginChunkFetcher:chunkFetcher offset:offset];
-                                  });
-                                }];
+
+    GTMSessionUploadFetcherDataProviderResponse response = ^(
+        NSData *chunkData, int64_t uploadFileLength, NSError *chunkError) {
```

Comment:
Should this get any queue assertion given the previous comments in the code about contexts?

Reply:
Done.